app/controllers/api/providers_controller.rb

```diff
@@ -61,6 +61,27 @@ def custom_attributes_add_resource(object, type, id, data = nil)
       super(object, type, id, formatted_data)
     end
 
+    def import_vm_resource(type, id = nil, data = {})
```

Comment:
Is there a reason to make `id` optional here? Does anyone call it without this argument?